### ISSUE TITLE
SpeechRecognizer.speechEndDetected does not adhere to value set in SpeechServiceConnection_EndSilenceTimeoutMs 

### ISSUE THREAD

**Issue by LedanDark** [Azure-Samples/cognitive-services-speech-sdk#505]
**Describe the bug**
In the Java SDK, SpeechRecognizer with a SpeechConfig with 
SpeechServiceConnection_EndSilenceTimeoutMs set to 1500 milliseconds takes less than 1500
milliseconds between recognized event handlers being called and speechEndDetected event handlers being called. 

From the Java documentation, I would expect this parameter to be the time between recognized and
speechEndDetected being called. https://docs.microsoft.com/en-us/java/api/com.microsoft.cognitiveservices.speech.propertyid?view=azure-java-stable

Reading the C# documentation, it is unclear if the endSilenceTimeout is the ambigous,unambigous endSilence timeout, or both. As mentioned in 
https://docs.microsoft.com/en-us/dotnet/api/system.speech.recognition.speechrecognitionengine.endsilencetimeout?view=netframework-4.8
https://docs.microsoft.com/en-us/dotnet/api/system.speech.recognition.speechrecognitionengine.endsilencetimeoutambiguous?view=netframework-4.8

**To Reproduce**
1. Create a SpeechConfig object with your credentials
2. config.setProperty(PropertyId.SpeechServiceConnection_EndSilenceTimeoutMs,1500)
3. Create a SpeechRecognizer object with this config.
4. Setup the SpeechRecognizer to log the time between recognized calls and speechEndDetected,
by adding eventhandlers to recognized and speechEndDetected.
5. Note that the time between calls is not ~1500 ms

**Expected behavior**
SpeechRecognizer object should wait close to SpeechServiceConnection_EndSilenceTimeoutMs 
parameter inbetween a call to recognized and senseSpeechEndDetected.

**Version of the Cognitive Services Speech SDK**
Java SDK 1.9.0

**Platform, Operating System, and Programming Language**
 - OS: Ubuntu 18.04
 - Hardware - x64
 - Programming language:  Java

**Additional context**
Is there other documentation for how the EndSilenceTimeout affects the recognition, or what
it is meant to do for the Java SDK?

**Comment by amitkumarshukla** [comment]
Thanks a lot for raising the issue. We are able to repro the issue locally. We are looking into it and will get back to you as soon as possible.

**Comment by amitkumarshukla** [comment]
We have found the issue and notified the right team and will keep you posted with the progress.

**Comment by rhurey** [comment]
Is it a typo in the repo steps that step 2 set the initial silence timeout and not the end one?

**Comment by LedanDark** [comment]
> Is it a typo in the repo steps that step 2 set the initial silence timeout and not the end one?

Ah, yes. Code is running with config.setProperty(SpeechServiceConnection_EndSilenceTimeoutMs,1500)

**Comment by amitkumarshukla** [comment]
Thanks a lot for the info. The following documentation do not belong to the Microsoft cognitive speech SDK. They are some older APIs.
https://docs.microsoft.com/en-us/dotnet/api/system.speech.recognition.speechrecognitionengine.endsilencetimeout?view=netframework-4.8

For C# we have this https://docs.microsoft.com/en-us/dotnet/api/microsoft.cognitiveservices.speech.propertyid?view=azure-dotnet document.

Looks like SpeechServiceConnection_EndSilenceTimeoutMs was added by mistake and it is also not documented well. We will think about deprecating this in the newer release.

Thanks a lot for your input. I am closing this issue. If you see similar issue, please report it back.

**Comment by LedanDark** [comment]
Wait. Is the answer that Microsoft cognitive speech SDK does not support modification of  silence timeouts in speech recognition calls?

**Comment by amitkumarshukla** [comment]
Currently no.

**Comment by amitkumarshukla** [comment]
You may want to read this thread for further understanding https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/502

One more thing to try out will be use continuous recognition https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/how-to-choose-recognition-mode?pivots=programming-language-csharp 

Set the value of SpeechServiceConnection_EndSilenceTimeoutMs and it might help you.

**Comment by LedanDark** [comment]
@amitkumarshukla Thanks to @rhurey 's response in https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/502 I was able to diagnose the problem I was having :
I was using my own stream to push audio data to the SpeechRecognizer.

**Comment by LedanDark** [comment]
@amitkumarshukla I was able to diagnose and fix the problem with @rhurey 's response in https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/502 :
I was using a stream created by  AudioInputStream.createPushStream() to send audio to SpeechRecognizer. I assumed that this would configure the streams AudioFormat to be the default mentioned in the docs, 16khz, 16 bit, mono channel. I also tried using AudioInputStream.createPushStream(AudioStreamFormat.getDefaultInputFormat()).
Both gave me results where the Initial Timeout was about 1/2 of the value I set.

By setting the audioformat manually :
AudioInputStream.createPushStream(AudioStreamFormat.getWaveFormatPCM(16000L,16,1))
I got the results I expected. 

AudioInputStream.createPushStream() and AudioStreamFormat.getDefaultInputFormat() don't use
16khz, 16 bits, mono. I suspect they create an AudioInputStream with the Audio format of 8khz, 16 bits, mono.
